Mathlib/Data/Set/Finite/Parity.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2024 Pim Otte. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pim Otte
+-/
+import Mathlib.Data.Set.Card
+import Mathlib.Data.Set.Finite
+
+/-!
+# Finite sets and parity of their cardinality
+
+This file contains lemmas concerning finite sets and parity of their cardinality.
+-/
+
+namespace Set.Finite
+
+variable {V : Type _} {s : Set V}
+
+lemma one_lt_ncard_of_nonempty_of_even (hs : Set.Finite s) (hn : Set.Nonempty s)
+    (he : Even (s.ncard)) : 1 < s.ncard := by
+  have : s.ncard ≠ 0 := by
+    intro h
+    rw [@Set.nonempty_iff_ne_empty] at hn
+    exact hn <| (Set.ncard_eq_zero hs).mp h
+  have : s.ncard ≠ 1 := by
+    intro h
+    simp [h] at he
+  omega
+
+lemma even_card_diff_pair [DecidableEq V] {x y : V} {u : Set V} (hu : Set.Finite u)
```

Comment:
My first instinct seeing this theorem is that it might be better stated in the other direction. That is, `Even u → Even (insert x (insert y u))`. Of course, this is pretty trivial as the latter set has cardinality `2` more than the former. I think the usefulness of this is contingent on whichever proof you want to use it on, and even then, I'd ask if the `insert` form can't be used instead.

Reply:
Given the arguments, I'd suggest two lemmas, relating Even s or Odd s to the opposite for ‘insert x s‘.
And maybe an additional that proves that y does not belong to insert x s iff...

Reply:
Fixed this:)